src/cpu/arm.rs

```diff
@@ -172,13 +172,19 @@ fn detect_features() -> u32 {
     features
 }
 
+#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
+fn detect_features() -> u32 {
+    1 << 6 // sha512
+}
```

Comment:
above in the `features!` macro invocation, we have:
```
    // Keep in sync with `ARMV8_SHA256`.
    "sha2" => SHA256 {
        mask: 1 << 4,
    },
```
Apparently in C and in Rust, the "sha3" Aarch64 CPU feature flag is overloaded for both SHA-3 and SHA-512 CPU features. So you should add an entry:
```
    // Keep in sync with `ARMV8_SHA512`.
    // "sha3" is overloaded for both SHA-3 and SHA512.
    "sha3" => SHA512 {
        mask: 1 << 6,
    },
```

Then you will not need to add a `detect_features()` because it will be statically detected on aarch64-apple-darwin:
```
$ rustc --print=cfg --target aarch64-apple-darwin | grep sha
target_feature="sha2"
target_feature="sha3"
# MSRV
$ rustc +1.61.0 --print=cfg --target aarch64-apple-darwin | grep sha
target_feature="sha2"
target_feature="sha3"
```

Also below, add an assertion analogous to const `_AARCH64_APPLE_TARGETS_EXPECTED_FEATURES` called `_AARCH64_APPLE_DARWIN_TARGETS_EXPECTED_FEATURES` that asserts that the "sha3" feature is enabled statically for aarch64-apple-darwin.

If you want to do the same for aarch64-apple-ios, then we have to use dynamic feature detection because aarch64-apple-ios doesn't enable it statically:
```
$ rustc +1.61.0 --print=cfg --target aarch64-apple-ios | grep sha
target_feature="sha2"
```
The BoringSSL implementation of this at https://boringssl.googlesource.com/boringssl/+/1e15682f1a4bb64c48b84884976a2b5c4201e878%5E%21/ demonstrates how to do that dynamic feature detection for iOS and also for other targets.